fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -348,6 +351,17 @@ public OlapTable(long id, String tableName, List<Column> baseSchema, KeysType ke
         this.tableProperty = null;
     }
 
+    @Override
+    public synchronized Optional<String> mayGetDatabaseName() {
```

Comment:
why `synchronized`?  Can not think of a situation of data race.

Plus, may I ask why to use `Optional<String>`? Wont it be staightforward to return null directly?

Reply:
@dirtysalt  Optional<String> is better than null,  the former would force coder to check if the value is null, the latter would not.